pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1822,21 +1730,23 @@ func (s *AWSCloud) updateInstanceSecurityGroupsForLoadBalancer(lb *elb.LoadBalan
 
 		permissions := []*ec2.IpPermission{permission}
 
-		if add {
-			changed, err := s.ensureSecurityGroupIngress(instanceSecurityGroupId, permissions)
+		if instanceSecurityGroupId != nil && groupId == instanceSecurityGroupId {
```

Comment:
Fixed derefs - I still need someone to tell me how to test this stuff :)

instanceSecurityGroupId is null when this method is called from EnsureTCPLoadBalancerDeleted.